### ISSUE TITLE
[bug]: Spend-Output monitoring might have a rare edge case where it misses the spend of an output.

### ISSUE THREAD

**Issue by ziggie1984** [lightningnetwork/lnd#8120]
There are reported cases (#8028) where a noderunner had invalid values in his height hint cache leading to lnd not recognizing already spent outputs as spent. Using the config setting  `--height-hint-cache-query-disable` in lnd, these situation were resolved. Maybe the historical recan logic has a race condition where it misses a spend somehow which eventually updates the height hint cache and therefore never recovering the monitored outputs because the height hint cache will be queried for any other historical rescan if we do not actively disable it. 

I was not able to reproduce this issue so far.

**Comment by Crypt-iQ** [comment]
Did this happen in one of the cases where bitcoind's rpc serial version was set to the legacy witness encoding?

**Comment by ziggie1984** [comment]
Yes the case I referenced had this setting on (rpcserialversion=0), but I am not sure how this could impact the monitoring because we are checking for outputs only ? Moreover after he set the `--height-hint-cache-query-disable` the historical rescan did work and lead to the situation where the preimage witness check happend (leading us to his problem of the `rpcserialversion` setting).
